Run kube-downscaler in test clusters

hjacobs opened the pull request:
Run [kube-downscaler](https://github.com/hjacobs/kube-downscaler) in all test clusters with some default configuration (to be overwritten via config items).

Needs internal documentation and announcement.

hjacobs committed:
run kube-downscaler in test clusters

hjacobs committed:
add serviceAccount with patch permissions

hjacobs committed:
v0.3 with grace period support

mikkeloscar commented on cluster/manifests/kube-downscaler/deployment.yaml:
```diff
@@ -0,0 +1,41 @@
+          - "--default-uptime={{ .ConfigItems.downscaler_default_uptime }}"
+        resources:
+          limits:
+            memory: 100Mi
```
Can we please avoid over-committing :)

hjacobs replied:
@mikkeloscar done, fixed --- this was not intended and I copied the manifest from https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/cluster/manifests/external-dns/deployment.yaml . We should probably once clean up all manifests if we want to avoid overcommit.

hjacobs committed:
avoid memory overcommit

hjacobs committed:
use image from Zalando OSS registry

muaazsaleem commented:
Since this is first of many changes to our test environment that won't affect prod clusters, should we also take this opportunity to create a "stable" channel for prod clusters? 
AFAIK we've already agreed to do it eventually but haven't decided on a time.

szuecs commented:
@muaazsaleem please have this a separate discussion from this PR. We had it already 1-2 times to discuss and we agreed on having more automations to support the channel flow, before creating more channels, but I think we can also discuss it again offline.

Commit:
Excluding postgres-operator deployment too

In addition to the deployments excluded by default also excluding "postgres-operator" deployments on user request.

szuecs commented:
:+1:

hjacobs commented:
:+1:
